modules/es-extensions/store/modules/greg-publisher-api.js

```diff
@@ -273,48 +273,24 @@ var gregAPI = {};
     gregAPI.userRegistry = function(session) {};
     gregAPI.assetManager = function(session, type) {};
     var assetManager = function(session, type) {
-        var rxt = require('rxt');
-        var am = rxt.asset.createUserAssetManager(session, type);
-        return am;
+        var tenantAPI = require('/modules/tenant-api.js').api;
+        var options = {'type':type};
+        var tenantResources = tenantAPI.createTenantAwareAssetResources(session, options);
+        return tenantResources.am;
     };
     /*Need assetManager for getAssetVersions*/
-    gregAPI.getAssetVersions = function (session, type, path, name) {
-        var versions = [];
-        var user = server.current(session);
-        if(!user){
-            return versions;
-        }
-        var am = assetManager(session, type);
-        var resource = am.registry.registry.get(path);
-        var params = path.split("/" + name);
-        var version_left_index = params[0];
-        var collection_path = version_left_index.substring(0, version_left_index.lastIndexOf("/"));
-        var base_version = version_left_index.substring(version_left_index.lastIndexOf("/") + 1, version_left_index.length);
-
-        var resource = am.registry.get(collection_path);
-        var children;
-        var collection;
-        if (resource.collection) {
-            collection = resource;
-        }
-
-        if (!resource.collection) {
-            throw 'Provided resource is not a collection';
-        }
+    gregAPI.getAssetVersions = function (session, type, path, name) { /* TODO: Instead of path accept asset itself*/
```

Comment:
Thanks @splinter , I have removed it in [this commit](https://github.com/wso2/product-greg/pull/854/commits/d36b5511870da574cce6377a0a15340d28de0b24)